### PR TITLE
Attempt at Auth :)

### PR DESCRIPTION
**Pull request by ukd1**
@siddontang this is my attempt at #182 :+1:

### CODE REVIEW

**Comment by ukd1** [comment]
FYI, this does not support auth for HTTP.

File: config/config.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ type SnapshotConfig struct {
 type Config struct {
 	m sync.RWMutex `toml:"-"`
 
+	AuthEnabled  bool   `toml:"auth_enabled"`
+	AuthPassword string `toml:"auth_password"`
```
Comment (siddontang) [inline config/config.go]: Maybe we only need AuthPassword, if it is not empty, we will use password, otherwise, don't. :-)

**Comment by siddontang** [comment]
Thanks @ukd1 

Not supporting HTTP is ok.

File: config/config.go (hunk 1 above)
Comment (ukd1) [inline config/config.go]: Sure, no problem.

**Comment by ukd1** [comment]
@siddontang that should be fixed; added some docs too :+1:

File: server/client.go [hunk 2]
```diff
@@ -64,6 +64,8 @@ type client struct {
 	cmd        string
 	args       [][]byte
 
+	is_authed bool
```
Comment (siddontang) [inline server/client.go]: The variable format is_authed is not suggested in Go. we could use isAuthed or other camel format instead. :-)

Comment (ukd1) [inline server/client.go]: Heheh ok - as mentioned, I'm a go newb.

**Comment by ukd1** [comment]
@siddontang ok, I've fixed the ``is_authed`` --> ``isAuthed`` :+1:

**Comment by siddontang** [comment]
Cool, Thank you very much! @ukd1

**Pull request merged**